service/exporter/k8s/k8s.go

```diff
@@ -0,0 +1,386 @@
+			metadata: metadata{
+				Labels: d.Labels,
+			},
+		})
+	}
+
+	return s, nil
+}
+
+func getPods(c *kubernetes.Clientset, ns string) []Pod {
+	// creates the in-cluster config
+	pods, err := c.Pods(ns).List(metav1.ListOptions{})
+	if err != nil {
+		//TODO use other than panic
+		panic(err.Error())
+	}
+
+	ps := []Pod{}
+	for _, p := range pods.Items {
+		ps = append(ps, Pod{
+			Name:              p.Name,
+			Status:            string(p.Status.Phase),
+			ContainerStatuses: p.Status.ContainerStatuses,
+			metadata: metadata{
+				Labels: p.Labels,
+			},
+		})
+	}
+	return ps
+}
+
+func getServices(c *kubernetes.Clientset, ns string) ([]Service, error) {
+	services, err := c.Services(ns).List(metav1.ListOptions{})
+	if err != nil {
+		return []Service{}, microerror.Mask(err)
+	}
+
+	ss := []Service{}
+	for _, s := range services.Items {
+		ss = append(ss, Service{
+			Name:     s.Name,
+			Ports:    s.Spec.Ports,
+			Type:     s.Spec.Type,
+			Selector: s.Spec.Selector,
+			metadata: metadata{
+				Labels: s.Labels,
+			},
+		})
+	}
+	return ss, nil
+}
+
+func getDaemonSets(c *kubernetes.Clientset, ns string) ([]DaemonSet, error) {
+	dss, err := c.DaemonSets(ns).List(metav1.ListOptions{})
+
+	if err != nil {
+		return []DaemonSet{}, microerror.Mask(err)
+	}
+
+	ds := []DaemonSet{}
+	for _, s := range dss.Items {
+
+		ds = append(ds, DaemonSet{
+			metadata: metadata{
+				Labels: s.Labels,
+			},
+			Status:      s.Status,
+			Selector:    *s.Spec.Selector,
+			PodTemplate: fromPodTemplateSpec(s.Spec.Template),
+		})
+	}
+	return ds, nil
+}
+
+func getStatefulSets(c *kubernetes.Clientset, ns string) ([]StatefulSet, error) {
+
+	ss := []StatefulSet{}
+	statefulSets, err := c.AppsV1beta1().StatefulSets(ns).List(metav1.ListOptions{})
+
+	if err != nil {
+		return ss, microerror.Mask(err)
+	}
+
+	for _, s := range statefulSets.Items {
+
+		ss = append(ss, StatefulSet{
+			metadata: metadata{
+				Labels: s.Labels,
+			},
+			ServiceName: s.Spec.ServiceName,
+			Replicas:    *s.Spec.Replicas,
+			PodTemplate: fromPodTemplateSpec(s.Spec.Template),
+			Selector:    *s.Spec.Selector,
+			Status:      s.Status,
+		})
+	}
+
+	return ss, nil
+}
+
+func getCronJobs(c *kubernetes.Clientset, ns string) ([]CronJob, error) {
+
+	cj := []CronJob{}
+	cronjobs, err := c.BatchV2alpha1().CronJobs(ns).List(metav1.ListOptions{})
+
+	if err != nil {
+		return cj, microerror.Mask(err)
+	}
+
+	for _, s := range cronjobs.Items {
+		cj = append(cj, CronJob{
+			metadata: metadata{
+				Labels: s.Labels,
+			},
+			Name:        s.GetName(),
+			Schedule:    s.Spec.Schedule,
+			Status:      s.Status,
+			Suspend:     *s.Spec.Suspend,
+			JobTemplate: fromJobTemplateSpec(s.Spec.JobTemplate),
+		})
+	}
+	return cj, nil
+}
+
+func getIngresses(c *kubernetes.Clientset, ns string) ([]Ingress, error) {
+	is := []Ingress{}
+	ingresses, err := c.ExtensionsV1beta1().Ingresses(ns).List(metav1.ListOptions{})
+
+	if err != nil {
+		return is, microerror.Mask(err)
```

Comment:
AFAIK golang convention is to return the zero value in case an error is returned. In this case this would be `nil`.